server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.common.hash.FNV1a;
+import org.opensearch.index.remote.RemoteStorePathStrategy.PathInput;
+
+import java.util.Locale;
+import java.util.Set;
+
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.DATA;
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.METADATA;
+
+/**
+ * This class contains the different enums related to remote store like data categories and types, path types
+ * and hashing algorithm.
+ *
+ * @opensearch.api
+ */
+public class RemoteStoreEnums {
```

Comment:
This class is just refactoring for moving all enums to one place. Additionally PathHashAlgorithm has been added.